project/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,161 @@
+package com.google.sps;
+
+import com.google.sps.data.Event;
+import com.google.sps.data.User;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class UserTest {
+  private static final String NAME = "Bob Smith";
+  private static final String EMAIL = "bobsmith@example.com";
+  private static final Set<String> INTERESTS =
+      Collections.unmodifiableSet(new HashSet<>(Arrays.asList("Conservation", "Food")));
+  private static final Set<String> SKILLS =
+      Collections.unmodifiableSet(new HashSet<>(Arrays.asList("Cooking")));
+  private static final String INTEREST_TO_ADD = "Music";
+  private static final String SKILL_TO_ADD = "Juggling";
+
+  // The following fields will be instantiated with meaningful Events after the Events class is
+  // written to follow the Builder pattern.
+  private static final Set<Event> EVENTS_HOSTING = Collections.emptySet();
+  private static final Set<Event> EVENTS_PARTICIPATING = Collections.emptySet();
+  private static final Set<Event> EVENTS_VOLUNTEERING = Collections.emptySet();
+
+  @Test
+  public void getNameAfterBuild() {
```

Comment:
I think you don't need so many tests since using Builder to create a User class is one behavior. We should test behavior, not function.

So I think you only need 2 tests:
  1. Use Builder to create one class, and verify all fields are correctly set.
  2. Use toBuilder to transfer a User to User.Builder, and verify all fields are transferred.

Reply:
Done, also added ones to check mutation since there are add/remove methods now